### PR TITLE
actions/api tests

### PR DESCRIPTION
**Pull request by na3d**

### CODE REVIEW

File: test/actions/api/datacenters.spec.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import sinon from 'sinon';
+import { expect } from 'chai';
+import {
+  UPDATE_DATACENTERS,
+  fetchDatacenters
+}from '../../../src/actions/api/datacenters';
```
Comment (SirCmpwn) [inline test/actions/api/datacenters.spec.js]: I've been favoring absolute imports (i.e. `~/actions/api/datacenters`) over using .. this many times

File: test/actions/api/datacenters.spec.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  it('should fetch datacenters', async () => {
+    await mockContext(sandbox, async ({
+        auth, dispatch, getState, fetchStub
+      }) => {
+      const f = fetchDatacenters();
+
+      await f(dispatch, getState);
+
+      expect(fetchStub.calledWith(
+        auth.token, '/datacenters?page=1')).to.be.true;
```
Comment (SirCmpwn) [inline test/actions/api/datacenters.spec.js]: `.to.equal(true)`

Comment (SirCmpwn) [inline test/actions/api/datacenters.spec.js]: (this will fail the build shortly)

File: test/actions/api/linodes.spec.js [hunk 3]
```diff
@@ -5,15 +5,74 @@ import {
...
+      expect(fetchStub.calledWith(
+        auth.token, '/linodes/linode_1')).to.be.true;
+      expect(dispatch.calledWith({
+        type: UPDATE_LINODE,
+        linode: mockUpdateLinode
+      })).to.be.true;
+    }, mockUpdateLinode);
+  });
 
+  it('should preform request update linode until condition is met');
```
Comment (SirCmpwn) [inline test/actions/api/linodes.spec.js]: Intend to implement this test in another PR?

File: test/actions/api/linodes.spec.js [hunk 4]
```diff
@@ -5,15 +5,74 @@ import {
...
+
+  it('should delete linode', async () => {
+    const f = deleteLinode('linode_1');
+
+    expect(f).to.be.a('function');
```
Comment (SirCmpwn) [inline test/actions/api/linodes.spec.js]: This doesn't test what it says on the tin. Make one test that asserts it returns a function, and another test that asserts it made the API call.

File: test/actions/authentication.spec.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+import {
+  SET_TOKEN,
+  setToken
+} from '../../src/actions/authentication';
+
+describe('actions/authentication', sinon.test(() => {
+  it('should return authentication', () => {
+    const f = setToken('token', 'scopes', 'username', 'email@domain.com');
+
+    expect(f.type).to.equal(SET_TOKEN);
```
Comment (SirCmpwn) [inline test/actions/authentication.spec.js]: This spec also doesn't make sense. It asserts that it returns a SET_TOKEN action, not that it returns "authentication". Should assert that it just applies the state as you expected. Also, use deepFreeze on all of your state objects when testing reducers, please.

File: test/components/Dropdown.spec.js [hunk 6]
```diff
@@ -4,7 +4,7 @@ import { mount } from 'enzyme';
 import { expect, assert } from 'chai';
 import Dropdown from '../../src/components/Dropdown';
 
-describe('Dropdown', () => {
+describe('components/Dropdown', () => {
```
Comment (SirCmpwn) [inline test/components/Dropdown.spec.js]: Do we want to establish this convention globally?

Comment (na3d) [inline test/components/Dropdown.spec.js]: I like the concept, since it makes it pretty clear where the tests live, as well as the piece being tested.

**Pull request closed without merging**